src/lib/actions/AbstractRunAction.ts

```diff
@@ -26,14 +29,17 @@ export abstract class AbstractRunAction implements Action {
 	private readonly inputProcessor: InputProcessor;
 	private readonly ruleFilterFactory: RuleFilterFactory;
 	private readonly engineOptionsFactory: EngineOptionsFactory;
+	private readonly resultsProcessorFactory: ResultsProcessorFactory;
 
 	protected constructor(logger: Logger, display: Display, inputProcessor: InputProcessor,
-							ruleFilterFactory: RuleFilterFactory, engineOptionsFactory: EngineOptionsFactory) {
+							ruleFilterFactory: RuleFilterFactory, engineOptionsFactory: EngineOptionsFactory,
+							resultsProcessorFactory: ResultsProcessorFactory) {
 		this.logger = logger;
 		this.display = display;
 		this.inputProcessor = inputProcessor;
 		this.ruleFilterFactory = ruleFilterFactory;
 		this.engineOptionsFactory = engineOptionsFactory;
+		this.resultsProcessorFactory = resultsProcessorFactory;
```

Comment:
Runtime dependencies should be created by the "main" function... which in our case is the files under commands/scanner . These action files can now be more easily unit tested because we inject the dependencies into them. Note that I don't call "new" inside of the RunAction or RunDfaAction class either... it is inside of run.ts and dfa.ts The fact we have an AbstractRunAction is simply implementation detail.